### PR TITLE
Add regression test for #70155.

### PR DESCRIPTION
**Pull request by CDirkx**
With #70166 merged, `RangeInclusive` now derives `PartialEq` and `Eq`, implementing structural equality and as a side effect the range is now usable with const generics, closing #70155.

As per [#70166 (comment)](https://github.com/rust-lang/rust/pull/70166#issuecomment-601872201) a test is added to avoid a change to the private fields or the equality implementation of the range from subtly reverting #70155.

### CODE REVIEW

**Comment by nikomatsakis** [comment]
Seems fine on technical grounds, but I'm going to make this

r? @oli-obk 

since it seems to indicate an (informal-ish) commitment and I want to get another set of eye-balls on that.

File: src/test/ui/const-generics/issues/issue-70155.rs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// check-pass
+#![allow(incomplete_features)]
+#![feature(const_generics)]
+
+// Regression test for #70155:
+// `RangeInclusive` should be usable with const generics
+
+struct S<const R: std::ops::RangeInclusive<usize>>;
```
Comment (oli-obk) [inline src/test/ui/const-generics/issues/issue-70155.rs]: Are there tests for the other range types? If so, please place this test with them, if not, add tests for them here, too.

**Comment by oli-obk** [comment]
> since it seems to indicate an (informal-ish) commitment and I want to get another set of eye-balls on that.

This is a general problem with structural equality being automatically derived together with equality if the fields support it. I think this change is fine, and it's good that all the range types behave the same now

File: src/test/ui/const-generics/issues/issue-70155.rs (hunk 1 above)
Comment (CDirkx) [reply src/test/ui/const-generics/issues/issue-70155.rs]: `RangeInclusive` is the only range type more complex than just a wrapper around a stable public `start` and/or `end` field, where the equality implementation might reasonably change. 

I thus initially only added this regression test for `RangeInclusive`, but can add them for the other ranges for completeness.

**Comment by oli-obk** [comment]
Please put all the tests into the same file. They are similar enough not to need a separate test file

**Comment by oli-obk** [comment]
@bors r+ rollup

Thanks!

**Pull request merged**